ElectronicObserver/Window/Tools/ExpeditionRecordViewer/ExpeditionRecordViewModel.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Linq;
+using ElectronicObserver.Database.Expedition;
+using ElectronicObserver.Database.KancolleApi;
+using ElectronicObserver.KancolleApi.Types;
+using ElectronicObserver.KancolleApi.Types.ApiReqMission.Result;
+using ElectronicObserver.Window.Tools.SortieRecordViewer;
+using ElectronicObserverTypes;
+using System.Text.Json;
+using System.Collections.Generic;
+using ElectronicObserver.Data;
+
+namespace ElectronicObserver.Window.Tools.ExpeditionRecordViewer;
+
+public class ExpeditionRecordViewModel
+{
+	public ExpeditionRecord Model { get; }
+	public int Id => Model.Id;
+	public DateTime ExpeditionStart { get; }
+	public IFleetData? Fleet { get; }
+	public List<int> ItemList { get; }
+	public object MaterialList { get; }
+	public int MaterialFuel { get; }
+	public int MaterialAmmo { get; }
+	public int MaterialSteel { get; }
+	public int MaterialBaux { get; }
+	public int? ItemOneID { get; }
+	public string ItemOneName { get; }
+	public int? ItemOneCount { get; }
+	public int? ItemTwoID { get; }
+	public int? MapAreaID { get; }
+	public string? DisplayID { get; }
+	public string ItemTwoName { get; }
+	public int? ItemTwoCount { get; }
+	public string ClearResult { get; }
+	public string ItemOneString => ItemList.Count > 0 && ItemOneCount > 0 ? ParseUseItem(ItemOneID, ParseResponse()!.ApiGetItem1?.ApiUseitemId) : "";
+	public string ItemTwoString => ItemTwoCount > 0 && ItemList.Count > 0 ? ParseUseItem(ItemTwoID, ParseResponse()!.ApiGetItem2?.ApiUseitemId) : "";
+	public ExpeditionRecordViewModel(ExpeditionRecord expedition, DateTime expeditionStart)
+	{
+		Model = expedition;
+		ExpeditionStart = expeditionStart.ToLocalTime();
+		MapAreaID = KCDatabase.Instance.Mission.Values.FirstOrDefault(s => s.MissionID == expedition.Expedition)?.MapAreaID;
+		DisplayID = KCDatabase.Instance.Mission.Values.FirstOrDefault(s => s.MissionID == expedition.Expedition)?.DisplayID;
+		Fleet = expedition.Fleet.MakeFleet();
+		ItemList = ParseResponse()!.ApiUseitemFlag;
```

Comment:
Don't reparse the response every time. Parse it once and save it as a local variable or a property.